### PR TITLE
WinAPI usage for service checks

### PR DESCRIPTION
**Pull request by sourque**

### CODE REVIEW

File: src/utility_windows.go [hunk 1]
```diff
@@ -242,3 +242,18 @@ func getLocalUser(userName string) (shared.LocalUser, error) {
...
+func getLocalServiceStatus(serviceName string) (shared.Service, error) {
+    serviceDataList, err := wapi.GetServices()
+    var serviceStatusData shared.Service
+	if err != nil {
+        failPrint("Couldn't get local service: " + err.Error())
+        return serviceStatusData, err
+    }
+    for _, v := range serviceDataList {
+        if v.SCName == serviceName {
+            serviceStatusData = v
```
Comment (sourque) [inline src/utility_windows.go]: Could probably just return the status at this point and save yourself some loops

File: src/checks_windows.go [hunk 2]
```diff
@@ -204,10 +216,50 @@ func packageInstalled(packageName string) (bool, error) {
...
+		var wantedStartupTypeNumber string
+		startupType = strings.ToLower(startupType)
+		if startupType == "automatic" {
+			wantedStartupTypeNumber = "2"
+		} else if startupType == "manual" {
+			wantedStartupTypeNumber = "3"
+		} else if startupType == "disabled" {
+			wantedStartupTypeNumber = "4"
+		} else {
+			return false, nil
```
Comment (sourque) [inline src/checks_windows.go]: Should return an error here for unknown startUp type (with a failPrint)

Comment (safinsingh) [reply src/checks_windows.go]: safinlint: unhandled error: try: 
```go
failPrint("Invalid startup type " + startupType)
return false, errors.New("Invalid startup type " + startupType)
```

File: src/utility_windows.go (hunk 1 above)
Comment (safinsingh) [reply src/utility_windows.go]: safinlint: what shane said: try:
```go
return serviceStatusData, nil
}
}
failPrint("Invalid service name: " + serviceName)
return nil, errors.New("Invalid service name: " + serviceName)
```

File: src/checks_windows.go [hunk 3]
```diff
@@ -204,10 +216,52 @@ func packageInstalled(packageName string) (bool, error) {
...
+	if err != nil {
+		return false, err
+	}
+	wantedStatus = strings.ToLower(wantedStatus)
+	if wantedStatus == "running" {
+		boolWantedStatus = true
+	} else if wantedStatus == "stopped" {
+		boolWantedStatus = false
+	} else {
+		errMessage := `Unknown startup type found for `+serviceName
```
Comment (safinsingh) [inline src/checks_windows.go]: spacing, extra use of `

File: src/checks_windows.go (hunk 2 above)
Comment (safinsingh) [reply src/checks_windows.go]: switch case?

File: src/checks_windows.go [hunk 4]
```diff
@@ -204,10 +216,52 @@ func packageInstalled(packageName string) (bool, error) {
...
+			wantedStartupTypeNumber = "2"
+		} else if startupType == "manual" {
+			wantedStartupTypeNumber = "3"
+		} else if startupType == "disabled" {
+			wantedStartupTypeNumber = "4"
+		} else {
+			return false, nil
+		}
+		check, err := registryKey(serviceKey, wantedStartupTypeNumber, false)
+		// check, err := commandOutput(`(Get-Service '`+serviceName+`').StartType`, startupType)
```
Comment (safinsingh) [inline src/checks_windows.go]: should prob delete this now

File: src/utility_windows.go [hunk 5]
```diff
@@ -242,3 +242,19 @@ func getLocalUser(userName string) (shared.LocalUser, error) {
 	}
 	return shared.LocalUser{}, nil
 }
+
+func getLocalServiceStatus(serviceName string) (shared.Service, error) {
+    serviceDataList, err := wapi.GetServices()
+    var serviceStatusData shared.Service
```
Comment (safinsingh) [inline src/utility_windows.go:248]: you can remove this entirely and set the failed values to nil

File: src/utility_windows.go [hunk 6]
```diff
@@ -242,3 +242,19 @@ func getLocalUser(userName string) (shared.LocalUser, error) {
...
+	if err != nil {
+        failPrint("Couldn't get local service: " + err.Error())
+        return serviceStatusData, err
```
Comment (safinsingh) [inline src/utility_windows.go:251]: return nil, err

File: src/utility_windows.go [hunk 7]
```diff
@@ -242,3 +242,19 @@ func getLocalUser(userName string) (shared.LocalUser, error) {
...
+    }
+    for _, v := range serviceDataList {
+        if v.SCName == serviceName {
+            return v, nil
+        }
+	}
+	failPrint(`Specified service '`+serviceName+`' was not found on the system`)
+	return serviceStatusData, err
```
Comment (safinsingh) [inline src/utility_windows.go:259]: return nil, err

**Review by safinsingh — CHANGES_REQUESTED** [review]

File: src/utility_windows.go (hunk 6 above)
Comment (Akshay-Rohatgi) [reply src/utility_windows.go:251]: cant return nil cus it's not shared.service type

File: src/utility_windows.go (hunk 7 above)
Comment (Akshay-Rohatgi) [reply src/utility_windows.go:259]: cant

File: src/utility_windows.go (hunk 5 above)
Comment (Akshay-Rohatgi) [reply src/utility_windows.go:248]: wdym

File: src/checks_windows.go (hunk 4 above)
Comment (Akshay-Rohatgi) [reply src/checks_windows.go]: will do

File: src/utility_windows.go (hunk 7 above)
Comment (safinsingh) [reply src/utility_windows.go:259]: y

File: src/checks_windows.go (hunk 3 above)
Comment (Akshay-Rohatgi) [reply src/checks_windows.go]: wat

File: src/utility_windows.go (hunk 6 above)
Comment (safinsingh) [reply src/utility_windows.go:251]: nil can be returned regardless of the return type

File: src/utility_windows.go (hunk 5 above)
Comment (safinsingh) [reply src/utility_windows.go:248]: see below

File: src/checks_windows.go (hunk 3 above)
Comment (safinsingh) [reply src/checks_windows.go]: errMessage := "Unknown startup type found for " + serviceName

File: src/utility_windows.go (hunk 6 above)
Comment (Akshay-Rohatgi) [reply src/utility_windows.go:251]: image

File: src/checks_windows.go [hunk 8]
```diff
@@ -204,10 +216,52 @@ func packageInstalled(packageName string) (bool, error) {
...
+		return false, err
+	}
+	switch wantedStatus = strings.ToLower(wantedStatus); wantedStatus {
+	case "running":
+		boolWantedStatus = true
+	case "stopped":
+		boolWantedStatus = false
+	default:
+		errMessage := "Unknown status type found for " + serviceName
+		failPrint(errMessage)
```
Comment (aaditgupta21) [inline src/checks_windows.go:236]: Can't you make this into one

Comment (Akshay-Rohatgi) [reply src/checks_windows.go:236]: i made it into a var so i dont have to put it twice

**Review by Akshay-Rohatgi — APPROVED** [review]
its too good i guess

**Review by safinsingh — APPROVED** [review]

**Pull request merged**